packages/itwin/tree-widget/README.md

```diff
@@ -585,62 +735,122 @@ Where `{tree}` specifies which tree component the feature is of.
+  <!-- BEGIN EXTRACTION -->
+
 ```tsx
-import { ModelsTreeComponent } from "@itwin/tree-widget-react";
+import { IModelContentTreeComponent } from "@itwin/tree-widget-react";
 
 function MyWidget() {
   return (
-    <ModelsTreeComponent
-      {...otherProps}
+    <IModelContentTreeComponent
       onPerformanceMeasured={(feature, elapsedTime) => {
-        console.log(`TreeWidget [${feature}] took ${elapsedTime} ms`)
+        console.log(`TreeWidget [${feature}] took ${elapsedTime} ms`);
       }}
       onFeatureUsed={(feature) => {
-        console.log(`TreeWidget [${feature}] used`)
+        console.log(`TreeWidget [${feature}] used`);
       }}
+      getSchemaContext={getSchemaContext}
+      selectionStorage={unifiedSelectionStorage}
     />
   );
 }
 ```
 
+  <!-- END EXTRACTION -->
+
 For custom tree components `TelemetryContextProvider` should be used:
 
+  <!-- [[include: [Presentation.Tree-widget.Telemetry-custom-tree-example-imports, Presentation.Tree-widget.Telemetry-custom-tree-example], tsx]] -->
+  <!-- BEGIN EXTRACTION -->
+
 ```tsx
-import { TelemetryContextProvider } from "@itwin/tree-widget-react";
+import {
+  TelemetryContextProvider,
+  TreeWithHeader,
+  useCategoriesTree,
+  useCategoriesTreeButtonProps,
+  VisibilityTree,
+  VisibilityTreeRenderer,
+} from "@itwin/tree-widget-react";
 
 function MyWidget() {
-  return <TelemetryContextProvider
-    componentIdentifier="MyTree"
-    onPerformanceMeasured={(feature, elapsedTime) => {
-      console.log(`TreeWidget [${feature}] took ${elapsedTime} ms`)
-    }}
-    onFeatureUsed={(feature) => {
-      console.log(`TreeWidget [${feature}] used`)
-    }}
-  >
-    <MyTree />
-  </TelemetryContextProvider>;
+  return (
+    <TelemetryContextProvider
+      componentIdentifier="MyTree"
+      onPerformanceMeasured={(feature, elapsedTime) => {
+        console.log(`TreeWidget [${feature}] took ${elapsedTime} ms`);
+      }}
+      onFeatureUsed={(feature) => {
+        console.log(`TreeWidget [${feature}] used`);
+      }}
+    >
+      <MyTree />
+    </TelemetryContextProvider>
+  );
 }
-
 function MyTree() {
-  // see "Custom trees" section for example implementation
+  const buttonProps = useCategoriesTreeButtonProps({ viewport });
+  const { categoriesTreeProps, rendererProps } = useCategoriesTree({ activeView: viewport, filter: "" });
+
+  return (
+    <TreeWithHeader
+      buttons={[
+        <CategoriesTreeComponent.ShowAllButton {...buttonProps} key={"ShowAllButton"} />,
+        <CategoriesTreeComponent.HideAllButton {...buttonProps} key={"HideAllButton"} />,
+      ]}
+    >
+      <VisibilityTree
+        {...categoriesTreeProps}
+        getSchemaContext={getSchemaContext}
+        selectionStorage={unifiedSelectionStorage}
+        imodel={imodel}
+        treeRenderer={(props) => <VisibilityTreeRenderer {...props} {...rendererProps} />}
+      />
+    </TreeWithHeader>
+  );
+  // see "Custom trees" section for more example implementations
```

Comment:
We could only include the Telemetry wrapper into the README and not show the `MyTree` component as it is shown in other examples.

Reply:
Now that I looked over the whole telemetry section, it seems this example is missing a piece. What's the point to provide telemetry context to a custom component? Probably so that the custom component could report used features and their performance, right? So we need `MyTree` to show that. Adding @saskliutas to confirm.

Reply:
Yes. `TelemetryContext` should be added if we want trees built using `VisibilityTree` or `Tree` component from tree-widget to report feature usage and performance,

Reply:
> Yes. `TelemetryContext` should be added if we want trees built using `VisibilityTree` or `Tree` component from tree-widget to report feature usage and performance,

Oh, so it's not like it's necessary for the custom component itself, but for the building blocks possibly used by the custom component (`VisibilityTree` or `Tree`), right? In that case, I think `MyTree` implementation should keep using `VisibilityTree` with the comment `// VisibilityTree will use provided telemetry context to report used features and their performance`. Also, I think the header part can be removed for simplicity.

Reply:
Yes. It is only necessary when using `VisibilityTree` or `Tree` component and you want them to report feature usage.

Reply:
Is the current version fine?

Reply:
LGTM